src/utilities/ScrollToListener/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { ReactElement, RefObject } from 'react';
+import { Props } from './types';
+
+
+/**
+ * Fires an function passed through onScrollTo when the child element is scrolled to
+ */
+function ScrollToListener({
+    children,
+    onScrollTo,
+    rootElement = null,
+    rootMargin = '0px 0px 0px 0px',
+    rootThreshold = 0.01,
+    willListen = false,
+}: Props): ReactElement {
+
+
+    const ref: RefObject<HTMLDivElement> = React.useRef(null);
+
+
+    function handleScroll(entries: any, observer: any): void {
+
+
+        const [entry] = entries;
+        const lastThreshold = observer.thresholds[observer.thresholds.length - 1];
+
+
+        if (entry.intersectionRatio > 0) onScrollTo();
+        if (!willListen && entry.intersectionRatio >= lastThreshold) observer.disconnect();
+
+
+    }
+
+
+    React.useEffect(() => {
+
+
+        const findRootElement = rootElement ? document.querySelector(rootElement) : null;
+
+        const target = ref.current?.previousElementSibling;
+
+        const observer = new IntersectionObserver(handleScroll, { root: findRootElement,
+            rootMargin,
+            threshold: rootThreshold });
+
+
+        if (target) observer.observe(target);
+
+
+        return ((): void => {
+            if (target) observer.unobserve(target);
+        });
+
+
+    }, []);
+
+
+    return (
+        <React.Fragment>
+            {children}
+            <div ref={ref} style={{ display: 'none' }} />
```

Comment:
One question - Should this div come before the children and use next sibling to get the element since we're observing something coming into view?  The only reason I ask is because I could see a case where someone uses a component with a react fragment like a list that outputs a bunch of sibling elements.  This selector would select the last one in that case, not the first one.

Reply:
Also, one minor thought - since you aren't using this element for anything but a reference, you might want to change it to a span tag instead of a div.  That way you don't get any invalid dom nesting errors if it gets put inside a paragraph tag, for example.

Reply:
@joshrcook Good catch, didn't think about that use case. Fixed.